portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SegmentsCreationWebBehavior.testcase

```diff
@@ -1373,4 +1373,55 @@ definition {
 		ACUtils.viewNameList(nameList = "ac ac");
 	}
 
+	@description = "Bug: LRAC-8233 | Automation ID: LRAC-11537 | Test Summary: Check web behavior shows which data source data came from"
+	@priority = "3"
+	test CheckWebBehaviorSegmentShowsWhichDataSourceDataCameFrom {
+		task ("Create a blog page and view page") {
+			ACUtils.createBlogsAndAddToPage();
+
+			ACDXPSettings.connectDXPtoAnalyticsCloud();
+
+			var dataSourceName = StringUtil.extractFirst("${assignedPropertyName}", "Combined");
+
+			var dataSourceName = StringUtil.trim("${dataSourceName}");
+
+			ACUtils.navigateToSitePage(= "AC Page", actionType = "View Blog", documentTitleList = "Blogs AC Title", siteName = "Site Name");
+
+			ACUtils.closeAllSessionsAndWait();
+		}
+
+		task ("Go to AC") {
+			ACUtils.launchAC();
+
+			ACProperties.switchProperty(propertyName = "${assignedPropertyName}");
+		}
+
+		task ("Create new Dynamic Segment") {
+			ACNavigation.goToSegments();
+
+			ACSegments.createDynamicSegment();
+
+			ACUtils.setItemName(itemName = "Dynamic Segment Test");
+
+			ACSegments.addSegmentField(segmentField = "Viewed Page");
+
+			ACSegments.editWebBehaviorCriterion(
+				indexField = "1",
+				occurenceNumber = "4",
+				searchTerm = "Blogs AC Title - Site Name - Liferay DXP");
+		}
+
+		task ("Check for data source origin") {
+			var key_indexField = "1";
+
+			Click(locator1 = "ACSegments#SELECT_BUTTON");
+
+			ACSegments.searchSegment(findSegment = "${documentTitle}");
+
+			ACUtils.viewGenericText(textValueList = "${dataSourceName}");
+
+			Click(locator1 = "ACSegments#MODAL_CANCEL_BUTTON");
```

Comment:
Nice job with the test! Just wondering if this step is necessary? Seems like you can end the test with the previous step.

Reply:
Hey @joshitagaki , yes for this case it's not necessary to cancel the modal. But as I mentioned above, this path can be use later on when we have to use a cancel button on popup modal for future tests.

Reply:
Ahh I see thanks! I didn't realize it was the same conversation haha.